lib/pdfkit/source.rb

```diff
@@ -38,11 +42,11 @@ def to_s
     private
 
     def shell_safe_url
-      url_needs_escaping? ? URI::escape(@source) : @source
+      url_needs_escaping? ? URI::DEFAULT_PARSER.escape(@source) : @source
```

Comment:
This is a direct fix of the `URI.escape` deprecation, and has the same exact behavior, but my main concern is that I don't think this is the right escaping method.

`URI.escape` escapes a string for use in a URI as a value. However, in this case, the string is being escaped for use in a shell command parameter, which doesn't really have the same requirements. For one thing, the percent-encoding that `URI.escape` generates might conflict with the `%var` environment variable syntax on Windows. There might be some other issues, but I haven't taken the time to check.

The change here has no effect on the existing behavior (except getting rid of the deprecation message), so it's probably the best to do here. However, in the long run, it's probably best to change to a more appropriate escaping method.